File: arrays/product_except_self/product_except_self.js
var productExceptSelf = function(nums) {
    const suffix = new Array(nums.length);
    suffix[nums.length - 1] = 1;
    i = nums.length - 2;
    while (i >= 0) {
        suffix[i] = suffix[i+1] * nums[i+1];
        i--;
    }
    prefix = [1];
    res = [suffix[0]];
    i = 1;
    while (i < nums.length) {
        prefix.push(prefix[i-1] * nums[i-1]);
        res.push(prefix[i] * suffix[i]);
        i++;
    }
    return res;
};

File: arrays/product_except_self/product_except_self.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* productExceptSelf(int* nums, int numsSize, int* returnSize){
    int suffix[numsSize];
    suffix[numsSize - 1] = 1;
    for (int i = numsSize - 2; i >= 0; i--) {
        suffix[i] = suffix[i+1] * nums[i+1];
    }
    int *res = malloc(numsSize * sizeof(int));
    *returnSize = numsSize;
    res[0] = suffix[0];
    int prefix[numsSize];
    prefix[0] = 1;
    for (int i = 1; i < numsSize; i++) {
        prefix[i] = prefix[i-1] * nums[i-1];
        res[i] = prefix[i] * suffix[i];
    }
    return res;
}